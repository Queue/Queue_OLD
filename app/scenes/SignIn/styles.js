//
// SignIn styles.js

import { StyleSheet, Dimensions } from 'react-native';
import { Colors } from '../../lib/globals';

let { width, height }  = Dimensions.get('window');

export default styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    width: width,
    height: height,
    backgroundColor: Colors.primaryBackground
  },
  wrapper: {
    width: 250
  }
});
