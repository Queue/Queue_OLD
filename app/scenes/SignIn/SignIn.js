//
// SignIn Scene

import React, { Component } from 'react';
import styles from './styles';
import {
  View,
  Text
} from 'react-native';
import {
  PrimaryButton,
  EmailField,
  PasswordField,
  TextButton
} from '../../components';
import Common from '../../lib/common';
import Data from '../../lib/data';
import KeyboardSpacer from 'react-native-keyboard-spacer';
import { Actions } from 'react-native-router-flux';

export default class SignIn extends Component {
  constructor(props) {
    super(props);

    this.state = {
      emailText: '',
      passwordText: ''
    };
  }

  signInUser() {
    let email = this.state.emailText,
        password = this.state.passwordText;

    if (password !== '' && Common.validateEmail(email)) {
      // Reset fields state
      this.setState({
        emailText: '',
        passwordText: ''
      });

     // sign in user
      Data.Auth.signIn(email, password).then(() => {
        Common.dismissKeyboard();
        Actions.DashboardRoute();
      }, (error) => {
        Common.error(error.code, error.message);
      });
    } else {
      Common.error('Error', 'Check your email and password.');
    }
  }

  render() {
    let email = this.state.emailText,
        password = this.state.passwordText;

    return (
      <View style = {styles.container}>
        <View style = {styles.wrapper}>
          <Text
            style = {styles.brand}>
            Queue
          </Text>
          <EmailField
            placeholder = {'Email'}
            change = {(text) => this.setState({emailText: text})}
            val = {email}
          />
          <PasswordField
            placeholder = {'Password'}
            secure = {true}
            change = {(text) => this.setState({passwordText: text})}
            val = {password}
          />
          <PrimaryButton
            name = {'Sign In'}
            press = {this.signInUser.bind(this)}
          />
          <TextButton
            text = {'Dont have an account?'}
            press = {Actions.SignUpRoute}
            size = {16}
          />
          <TextButton
            text = {'Forgot your password?'}
            press = {Actions.ForgotRoute}
            size = {12}
          />
          <KeyboardSpacer />
        </View>
      </View>
    );
  }

}
