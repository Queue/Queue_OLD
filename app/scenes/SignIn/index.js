//
// SignIn index.js

import SignIn from './SignIn';
export default SignIn;
