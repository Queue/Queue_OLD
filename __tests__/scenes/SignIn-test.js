//
// SignIn scene test

import React from 'react';
import SignIn from '../../app/scenes/SignIn';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <SignIn />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
