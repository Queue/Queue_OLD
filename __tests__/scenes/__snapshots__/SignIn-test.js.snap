exports[`test renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#eaeaea",
      "flex": 1,
      "height": 1334,
      "justifyContent": "center",
      "width": 750,
    }
  }>
  <View
    style={
      Object {
        "width": 250,
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#4AC948",
          "fontFamily": "airstream",
          "fontSize": 100,
          "marginBottom": -20,
          "textAlign": "center",
        }
      }>
      Queue
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 2,
          "height": 40,
          "justifyContent": "center",
          "margin": 10,
          "padding": 15,
        }
      }>
      <TextInput
        onChangeText={[Function]}
        placeholder="Email"
        style={
          Object {
            "flex": 1,
          }
        }
        value="" />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 2,
          "height": 40,
          "justifyContent": "center",
          "margin": 10,
          "padding": 15,
        }
      }>
      <TextInput
        onChangeText={[Function]}
        placeholder="Password"
        value="" />
    </View>
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits="button"
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderColor": "#4AC948",
          "borderRadius": 4,
          "borderWidth": 1,
          "margin": 15,
          "marginTop": 20,
          "opacity": 1,
          "overflow": "hidden",
          "padding": 8,
        }
      }
      testID={undefined}>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007aff",
              "fontFamily": "HelveticaNeue-Medium",
              "fontSize": 17,
              "fontWeight": "bold",
              "textAlign": "center",
            },
            null,
            Object {
              "color": "#4AC948",
              "fontSize": 20,
              "fontWeight": "300",
            },
            null,
          ]
        }>
        Sign In
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#4AC948",
            "fontSize": 14,
            "padding": 5,
            "textAlign": "center",
          }
        }>
        Dont have an account?
      </Text>
    </View>
  </View>
</View>
`;
